astropy/table/index.py

```diff
@@ -398,9 +398,8 @@ def __deepcopy__(self, memo):
         ----------
         memo : dict
         '''
-        num_cols = self.data.num_cols if self.engine == SortedArray else None
-        # create an actual Index, not a SlicedIndex
-        index = super(Index, Index).__new__(Index)
+        # Bypass Index.__new__ to create an actual Index, not a SlicedIndex.
+        index = super().__new__(self.__class__)
```

Comment:
I'm not 100% sure about this either, like whether everything would actually work for an `Index` subclass given that it is currently hardwired to `Index`.